Fix croniter_range infinite loop

snapiri opened the pull request:
The issue was caused due to the fact that the result time was adjusted unconditionally
though it might have resulted in datetime that does not really exist.

e.g.
If the clock was moved 1 hour forward at March 27 00:00, we do not have that time,
but rather March 27 01:00. If we adjust it we get March 26 23:00.

Closes: #20

kiorky commented:
Im gonna merge this as soon as travis is green (its already fine, there were spurious unrelaated timeouts)

snapiri committed:
Fix croniter_range infinite loop

The issue was caused due to the fact that the result time was adjusted unconditionally
though it might have resulted in datetime that does not really exist.
e.g.
If the clock was moved 1 hour forward at March 27 00:00, we do not have that time,
but rather March 27 01:00. If we adjust it we get March 26 23:00.

snapiri committed:
Add test
